pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -231,14 +231,19 @@ func (r *ReconcilePlanExecution) Reconcile(request reconcile.Request) (reconcile
 			planExecution.Spec.Instance.Name,
 			planExecution.Spec.Instance.Namespace,
 			err)
-		return reconcile.Result{}, err
+		if errors.IsNotFound(err) {
+			// do not retry when instance is not found, it's not probable that we can recover from that
+			return reconcile.Result{}, nil
+		} else {
+			return reconcile.Result{}, err
+		}
 	}
 
 	// Check for Suspend set.
 	if planExecution.Spec.Suspend != nil && *planExecution.Spec.Suspend {
 		planExecution.Status.State = kudov1alpha1.PhaseStateSuspend
 		err = r.Update(context.TODO(), planExecution)
-		r.recorder.Event(instance, "Normal", "PlanSuspend", fmt.Sprintf("PlanExecution %v suspended", planExecution.Name))
+		r.recorder.Event(instance, "Normal", "PlanSuspend", fmt.Sprintf("PlanExecution %v on instance %v suspended", planExecution.Name, planExecution.Spec.Instance.Name))
```

Comment:
Should we have a set of constants for the different types of Plan events?

Reply:
Nit: we're recording an event for an instance object. I don't know what's the common pattern for other k8s controllers, but including the instance name in the event message seems a bit redundant.